desktop/apps/partner2/components/fixed_cells_count_carousel/view.coffee

```diff
@@ -29,8 +29,10 @@ module.exports = class FixedCellsCountCarousel extends Backbone.View
       _.map @fetchOptions, (options) =>
         @collection.fetch data: options, remove: false
     ).then (values) =>
-      objects = [].concat.apply [], _.pluck values, 'value'
-      @collection.reset objects
+      objects = _(values).chain().pluck('value')
+      if objects.first().has('results').value() # For articles wrapped in results
```

Comment:
The naming here is kinda confusing, `values` is actually a bunch of promises from `Q.allSettled` which I then pluck out the data and wrap in underscore.js. What that data looks like is `[apiResponse, apiResponse]`—so yes it shouldn't be the case that an API returns null instead of empty array, but even if it does underscore is defensive against that, e.g.

```javascript
_.has(null, 'results') => false
_([null]).chain().first().has('results').value() => false
```

Thankfully with native promises (especially combined with [async/await](https://ponyfoo.com/articles/understanding-javascript-async-await)), this stuff becomes a lot more consistent + obvious.